maestro-utils/src/test/kotlin/SocketUtilsTest.kt

```diff
@@ -0,0 +1,66 @@
+import maestro.utils.SocketUtils
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import java.net.Inet4Address
+import java.net.InetAddress
+import java.net.NetworkInterface
+
+class SocketUtilsTest {
+
+    @Test
+    fun `nextFreePort should return a free port within the specified range`() {
+        val from = 5000
+        val to = 5100
+        val port = SocketUtils.nextFreePort(from, to)
+
+        assertTrue(port in from..to)
+    }
+
+    @Test
+    fun `nextFreePort should throw IllegalStateException when no ports are available in the range`() {
+        val from = 100000
+        val to = 100010
+
+        assertThrows(IllegalStateException::class.java) {
+            SocketUtils.nextFreePort(from, to)
+        }
+    }
+
+    @Test
+    fun `localIp should return a non-loopback IPv4 address`() {
+        val ip = SocketUtils.localIp()
+
+        assertNotNull(ip)
+        assertTrue(ip.startsWith("192") || ip.startsWith("10") || ip.startsWith("172") || ip.startsWith("127"))
+
+        val inetAddress = InetAddress.getByName(ip)
+
+        assertTrue(inetAddress is Inet4Address)
+        assertFalse(inetAddress.isLoopbackAddress)
```

Comment:
This fails if the returned address is a loopback address. What is the goal of this test? `SocketUtils.localIp` fallbacks to a loopback address so testing `localIp should return a non-loopback IPv4 address` doesn't really make sense to me so would love some context here?

Reply:
You are sure @axelniklasson! I reviewed this test and saw that it does not bring any changes to test coverage. When creating this test, my objective was to validate that the connection returned was not of the loopback type, but in fact this validation makes no sense. I refactored the tests related to the localIp method, making the objectives of each test clearer and even achieving greater test coverage. You can check commit https://github.com/mobile-dev-inc/maestro/pull/2081/commits/0378f46ec6b2c06f68841a1070b480d910d3b55c